### PR TITLE
feat: update resource key role

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Pass the resource key role as a module input

### Types of changes in this PR
https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/issues/206
#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Review by akocbek — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -60,6 +60,17 @@ variable "logdna_manager_key_name" {
   default     = "LogDnaManagerKey"
 }
 
+variable "logdna_resource_key_role" {
+  type        = string
+  description = "Role assigned to provide the IBM Cloud Logging key."
+  default     = "Manager"
+
+  validation {
+    condition     = contains(["Manager", "Reader", "Standard Member"], var.logdna_resource_key_role)
```
Comment (daniel-butler-irl) [inline variables.tf:69]: I am not familiar with the Standard Member role. Could you please provide a link to the docs where you saw it, thanks?

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]
Have all the roles been manually validated? Can you see the logs working when each is set?

File: variables.tf (hunk 1 above)
Comment (iamar7) [reply variables.tf:69]: The service roles for IBM log analysis as per [this](https://cloud.ibm.com/iam/roles) are Manager, Reader and Standard Member. As well as this [doc](https://cloud.ibm.com/docs/log-analysis?topic=log-analysis-iam#service) also gives information about service roles.

**Comment by iamar7** [comment]
> Have all the roles been manually validated? Can you see the logs working when each is set?

Yes, I have validated the manager and standard member role. 
Links for the logs -
- [Manager](https://cloud.ibm.com/observe/embedded-view/logging/8c4822bc-da0e-4989-a815-51bc8dafddc7)
- [Standard Member](https://cloud.ibm.com/observe/embedded-view/logging/4e2eb8a0-7e1a-4ab7-b107-450f935065ee)

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**